src/utils/cspUtils.js

```diff
@@ -0,0 +1,133 @@
+  return toml.parse(tomlFile.data);
+};
+
+export async function getCSP(repoName) {
+  const tomlData = await _parseNetlifyToml(repoName);
+  const csp = tomlData.headers[0].values['Content-Security-Policy'];
+  return csp;
+};
+
+function _stringContainsValue(string, value) {
+  const VALUE_REGEX = `(\\s|^)${value}(\\s|$)`;
+  const ValueRegexTest = new RegExp(VALUE_REGEX);
+  return ValueRegexTest.test(string);
+};
+
+function _toRegExp(string) {
+  const strippedString = string.replace(/\/$/, '') // removes ending '/' from domains eg 'abc.com/'
+  const escapedStrippedString = escapeStringRegexp(strippedString) // add escape characters 
+  const wildcardEscapedStrippedString = escapedStrippedString.replace('\\*', '[a-z0-9-]*') // convert wildcards
+  const RegExpString = RegExp(wildcardEscapedStrippedString) // converts to RegExp
+  return RegExpString;
+};
+
+/* Helper functions to check if elemSrc satisfies each CSP source specification: host-source, schema-source and 'self' */
+function _checkHostsourcePolicy (elemSrc, policy) {
+  if (_stringContainsValue(policy, '\\*')) return true;
+
+  const specialValues = ['http:', 'https:', 'data:', 'mediastream:', 'blob:', 'filesystem:', "'self'", "'none'", '*'];
+  const hostsources = policy.split(' ').filter(value => (!specialValues.includes(value)));
+  
+  const hostsourcesSatisfied = hostsources.some(hostsource => (_toRegExp(hostsource).test(elemSrc)));
+  return hostsourcesSatisfied;
+};
+
+function _checkSchemasourcePolicy (elemSrc, policy) {
+  const schemes = ['http:', 'https:'];
+  const dataschemes = ['data:', 'mediastream:', 'blob:', 'filesystem:'];
+
+  const schemesSatisfied = schemes.some(scheme => (_stringContainsValue(policy, scheme) && _.startsWith(elemSrc, scheme)) );
+  const dataschemesSatisfied = dataschemes.some(datascheme => (_stringContainsValue(policy, datascheme) && _.startsWith(elemSrc, datascheme)) );
+  return schemesSatisfied || dataschemesSatisfied;
+};
+
+function  _checkSelfPolicy (elemSrc, policy) {
+  return ( isLinkInternal(elemSrc) && _stringContainsValue(policy, "'self'"));
+};
+
+/* Helper function to check if elemAttr satisfies CSP source specifications */
+function _elemAttrSatisfiesPolicies (elemAttr, policy) {
+  if (_stringContainsValue(policy, "'none'")) return false; 
+  
+  const selfSatisfied = _checkSelfPolicy(elemAttr, policy);
+  const schemasourceSatisfied = _checkSchemasourcePolicy(elemAttr, policy);
+  const hostsourceSatisfied = _checkHostsourcePolicy(elemAttr, policy);
+  return selfSatisfied || schemasourceSatisfied || hostsourceSatisfied;
+};
+
+/* Helper function to get resource policy from csp for given policyType*/
+function _getResourcePolicy(csp, policyType) {
+  const resourcePolicyMapping = { 
+    'frame-src': ['frame', 'iframe'],
+    'img-src': ['img'],
+    'media-src': ['audio', 'video', 'track'],
+    'object-src': ['object', 'embed', 'applet'],
+    'script-src-elem': ['script'],
+  };
```

Comment:
Since this variable never changes, can we put this as a global variable?

Reply:
That's possible, but since it's only ever used in this function would it be tidier for this variable to only be in the scope of this function?

Reply:
Sure thing. If this turns out to affect performance down the line we can review it again